### ISSUE TITLE
Invalid image name on v1.23.1

### ISSUE THREAD

**Issue by michaelkourlas** [microsoft/vscode-docker#3752]
I'm experiencing an issue similar to #3719 in v1.23.1.

Output of `docker image ls --format "{{ json . }}"` on my remote SSH host:

```
{"Containers":"N/A","CreatedAt":"2022-07-05 13:43:31 -0400 EDT","CreatedSince":"5 months ago","Digest":"\u003cnone\u003e","ID":"a80cfbd6e315","Repository":"apps-jenkins:18888/devserver-centos8","SharedSize":"N/A","Size":"5.39GB","Tag":"1.0.4","UniqueSize":"N/A","VirtualSize":"5.39GB"}
```

Please let me know if you need other information.

**Comment by bwateratmsft** [comment]
Sort of copying my comment from #3719 but I'll include it here for posterity. I think that what's happening is that the image registry is sort of a top-level domain (`apps-jenkins`), and our code isn't expecting any top-level domains as a registry _except_ for `localhost` specifically.

**Comment by bwateratmsft** [comment]
We based our parsing on [this StackOverflow answer](https://stackoverflow.com/a/37867949). I think we misread the requirements slightly--upon rereading it again, it sounds like a registry hostname _can_ be a top-level domain, as long as a `:` port separator is included. In any case, it's clear that Docker accepted this as a valid image name--else it wouldn't have been tagged this way in the first place--so we need to update our parsing logic.

**Comment by danegsta** [comment]
A fix for how we're parsing image names has been released in 1.23.3.
